src/Core/Framework/Plugin/PluginLifecycleService.php

```diff
@@ -205,10 +205,18 @@ public function uninstallPlugin(
             $this->assetInstaller->removeAssetsOfBundle($pluginBaseClassString);
         }
 
+        if (!$uninstallContext->keepUserData() && Feature::isActive('v6.7.0.0')) {
+            // plugin->uninstall() will remove the tables etc of the plugin,
+            // we drop the migrations before, so we can recover in case of errors by rerunning the migrations
+            $pluginBaseClass->removeMigrations();
+        }
+
         $pluginBaseClass->uninstall($uninstallContext);
 
         if (!$uninstallContext->keepUserData()) {
-            $pluginBaseClass->removeMigrations();
+            if (!Feature::isActive('v6.7.0.0')) {
+                $pluginBaseClass->removeMigrations();
```

Comment:
Why do we do it before and after?

Reply:
we do it before when 6.7 flag is active, and after only for 6.6 (when 6.7 flag is **not/!** active)

I was not sure if that change might lead to problems for plugins, therefore I put that change behind a feature flag

Reply:
ahh yeah I get it now (missed the !). Thanks!